test.cmake

```diff
@@ -50,14 +50,20 @@ MACRO(ADD_UNIT_TEST NAME SOURCE)
 ENDMACRO(ADD_UNIT_TEST NAME SOURCE)
 
 #.rst:
-# .. command:: ADD_PYTHON_UNIT_TEST (NAME SOURCE)
+# .. command:: ADD_PYTHON_UNIT_TEST (NAME SOURCE [MODULE_PATH])
 #
-#   Add a test called `NAME` that runs an equivalent of ``python ${SOURCE}``
+#   Add a test called `NAME` that runs an equivalent of ``python ${SOURCE}``,
+#   optionnaly with a `PYTHONPATH` set to `CMAKE_BINARY_DIR/MODULE_PATH` if `MODULE_PATH` is set.
+#   `SOURCE` is relative to `PROJECT_SOURCE_DIR`
 #
 #   .. note:: :command:`FINDPYTHON` should have been called first.
 #
 MACRO(ADD_PYTHON_UNIT_TEST NAME SOURCE)
-  ADD_TEST(${NAME} ${PYTHON_EXECUTABLE} "${SOURCE}")
+  SET(MODULE_PATH "${ARGN}")  # ARGN is not a variable
+  ADD_TEST(${NAME} ${PYTHON_EXECUTABLE} "${PROJECT_SOURCE_DIR}/${SOURCE}")
+  IF(MODULE_PATH)
+    SET_PROPERTY(TEST ${NAME} PROPERTY ENVIRONMENT "PYTHONPATH=${CMAKE_BINARY_DIR}/${MODULE_PATH}:$ENV{PYTHONPATH}")
```

Comment:
A simple remark: when `$ENV{PYTHONPATH}` is empty, PYTHONPATH will be terminated by `some_path:` which might not be desired.

Reply:
The documentation says that «Non-existent directories are silently ignored.», and I thought that the directory with no name is non-existent… But it turns out an empty string is like `.` :/